enhancements/split-sdk-into-multiple-projects.md

```diff
@@ -0,0 +1,190 @@
+---
+title: split-sdk-into-multiple-projects
+authors:
+  - "@jesusr"
+reviewers:
+  - "@estroz"
+  - "@hasbro17"
+  - "@joelanford"
+  - "@fabianvf"
+  - "@shawn-hurley"
+approvers:
+  - "@estroz"
+  - "@hasbro17"
+  - "@joelanford"
+  - "@fabianvf"
+  - "@shawn-hurley"
+creation-date: 2020-05-18
+last-updated: 2020-05-27
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# Split SDK into multiple projects
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [X] Design details are appropriately documented from clear requirements
+- [X] Test plan is defined
+- [X] Graduation criteria for dev preview, tech preview, GA
+
+## Open Questions [optional]
+
+1. How will separate repos affect the SDK website and docs in general?
+   * What if each individual repo maintains their own docs? Then have a
+     pipeline sort of thing that would regenerate the entire doc if there are
+     any changes
+     * Could this be an intern project?
+   * How do versions work? netlify handles it
```

Comment:
IMO the repos will be GO libs. IN this way, we should only have the Go docs which are gen automatically based in the comments made in the code. Also, we should ensure it by a lint check in the CI. IMO it would be done as it is for controller-runtime. E.g Lint : https://github.com/kubernetes-sigs/controller-runtime/pull/885 and docs https://godoc.org/sigs.k8s.io/controller-runtime

See that any change in the LIB which is, for example, a breaking change will force us to do changes in the SDK CLI tool which is using it and then, will force us to update the SDK website/guide docs. So, I do not think that it would have any downside over keep the website docs updated accordingly. 

WDYT @joelanford @asmacdo @jmrodri ?

Reply:
@camilamacedo86 yeah that makes sense regarding the go libs. They won't need the docs. Makes sense.